lib/iris/analysis/__init__.py

```diff
@@ -1026,18 +1026,35 @@ def _percentile(data, axis, percent, **kwargs):
     If a new additive dimension is formed, then it will always be the last
     dimension of the resulting percentile data payload.
 
+    Kwargs:
+
+    * percentile_method (string) :
+        Switch to opt between the scipy.stats.mstats.mquantiles method for
+        calculating percentiles, and the much faster np.percentiles method.
+
     """
+    percentile_method = kwargs.get('percentile_method', None)
     # Ensure that the target axis is the last dimension.
     data = np.rollaxis(data, axis, start=data.ndim)
-    quantiles = np.array(percent) / 100.
     shape = data.shape[:-1]
     # Flatten any leading dimensions.
     if shape:
         data = data.reshape([np.prod(shape), data.shape[-1]])
     # Perform the percentile calculation.
-    result = scipy.stats.mstats.mquantiles(data, quantiles, axis=-1, **kwargs)
+    if percentile_method == 'fast':
+        msg = 'Cannot use fast np.percentile method with masked array.'
+        if ma.isMaskedArray(data):
+            raise TypeError(msg)
+        result = np.percentile(data, percent, axis=-1)
```

Comment:
does `np.percentile` take the same `**kwargs` as `scipy.stats.mstats.mquantiles`?

if it is, then I think the kwargs should be passed down

if not, then some thought about how to handle kwargs differently for different implementations should be considered